lib/components/AlbumScreen/track_list_tile.dart

```diff
@@ -626,6 +626,59 @@ class TrackListItemTile extends ConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final highlightTrack = isCurrentTrack && highlightCurrentTrack;
     final isOnDesktop = Platform.isMacOS || Platform.isWindows || Platform.isLinux;
+    final isOffline = ref.watch(finampSettingsProvider.isOffline);
+    final additionalBaseItemInfos = ref.watch(finampSettingsProvider.additionalBaseItemInfo);
+    final additionalBaseItemInfo =
+        (isOffline &&
+            (additionalBaseItemInfos[BaseItemDtoType.track] == AdditionalBaseItemInfoTypes.dateLastPlayed ||
+                additionalBaseItemInfos[BaseItemDtoType.track] == AdditionalBaseItemInfoTypes.playCount))
+        ? AdditionalBaseItemInfoTypes.none
+        : additionalBaseItemInfos[BaseItemDtoType.track];
+
+    final additionalInfo = switch (additionalBaseItemInfo) {
+      AdditionalBaseItemInfoTypes.playCount => (
+        playCount: true,
+        releaseDate: false,
+        dateAdded: false,
+        dateLastPlayed: false,
+      ),
```

Comment:
this also seems quite verbose, especially since you're dissecting it into separate variables at the end anyway...